src/azure-cli-core/azure/cli/core/commands/progress.py

```diff
@@ -0,0 +1,180 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from __future__ import division
+import sys
+from enum import Enum
+
+import humanfriendly
+
+BAR_LEN = 70
+
+
+class ProgressType(Enum):  # pylint: disable=too-few-public-methods
+    """ the types of progress """
+    Determinate = 0
+    Indeterminate = 1
+    # for views that want to accept both determinate and indeterminate progres
+    Both = 2
+
+
+class _ProgressViewBase(object):
+    """ a view base for progress reporting """
+    def __init__(self, out, progress_type, format_percent=None):
+        self.out = out
+        assert isinstance(progress_type, ProgressType)
+        self.progress_type = progress_type
+        self.format_percent = format_percent
+
+    def write(self, args):
+        """ writes the progress """
+        pass
+
+    def flush(self):
+        """ flushes the message out the pipeline"""
+        self.out.flush()
+
+    def get_type(self):
+        """ returns the progress type """
+        return self.progress_type
+
+
+class DeterminateProgressView(_ProgressViewBase):
+    """ a view base for progress reporting """
+    def __init__(self, out, format_percent=None):
+        super(DeterminateProgressView, self).__init__(out, ProgressType.Determinate, format_percent)
+
+
+class InDeterminateProgressView(_ProgressViewBase):
+    """ a view base for progress reporting """
+    def __init__(self, out, format_percent=None):
+        super(InDeterminateProgressView, self).__init__(
+            out, ProgressType.Indeterminate, format_percent)
```

Comment:
Remove these two classes. They do not define new behaviors. Avoid deep and unnecessary class inheritance.